.github/workflows/package-helm.yaml

```
@@ -30,10 +38,68 @@ jobs:
             mv $SRC_FILE $DEST_FILE
           done
 
-      - name: Upload Helm packages to GitHub releases
+      - name: Upload Stable Helm Charts to GitHub Release
         uses: softprops/action-gh-release@v1
         with:
           name: ${{ github.ref_name }}
           draft: true
           append_body: true
           files: nri-*helm-chart*.tgz
+
+  unstable:
+    concurrency:
+      group: unstable-helm-charts
+      cancel-in-progress: false
+    if: ${{ github.env.IS_RELEASE != 'true' }}
+    permissions:
+      packages: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Deep Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Install Helm
+        uses: azure/setup-helm@v4.0.0
+
+      - name: Package Unstable Helm Charts
+        id: package-charts
+        run: |
+          # For unstable chart verison we use:
+          #   - chart version: latest tag+1 patch + '-unstable'
+          #   - image version: 'unstable'.
+          majminpatch="$(git describe | sed -E 's/v([0-9]*\.[0-9]*\.[0-9]*).*$/\1/')"
```

Should we include the `v` here?

```suggestion
          majminpatch="$(git describe | sed -E 's/(v[0-9]*\.[0-9]*\.[0-9]*).*$/\1/')"
```

I.e. make the version `v0.4.0-unstable` instead of `0.4.0-unstable`.

---

Unfortunately, that's not a valid version for a Helm chart. It needs to be semver-compatible, so it can't start with a 'v'.

---

Although you are right that probably we could use a 'v' in the OCI registry tag and omit it from the version in the chart metadata... I can try it out in practice, and see if Helm is happy with such a setup...

---

Hmm, but we don't get to chose the tag or anything. Helm figures all that out by itself, so there is not much I can do here, I think.

---

But wait, how are the stable helm charts versioned starting with v:

```
$ helm search repo nri-plugins/nri-resource-policy-balloons -l
NAME                                    	CHART VERSION	APP VERSION	DESCRIPTION
nri-plugins/nri-resource-policy-balloons	v0.4.1       	v0.4.1     	The balloons NRI resource policy plugin impleme...
nri-plugins/nri-resource-policy-balloons	v0.3.2       	v0.3.2     	The balloons NRI resource policy plugin impleme...
...
```

---

Updated to use 'v'-prefixed versions and to push helm charts to `ghcr.io/$org/$repo/helm-charts/$helmchart`.